diff -Naur busybox-1.30.1-orig/procps/kill.c busybox-1.30.1/procps/kill.c
--- busybox-1.30.1-orig/procps/kill.c	2019-10-16 12:55:37.442215511 -0400
+++ busybox-1.30.1/procps/kill.c	2019-10-16 12:57:05.430128776 -0400
@@ -32,10 +32,10 @@
 //config:	in its own session, so it won't kill the shell that is running
 //config:	the script it was called from.
 
-//applet:IF_KILL(    APPLET_NOFORK(kill,     kill, BB_DIR_BIN,      BB_SUID_DROP, kill))
+//applet:IF_KILL(    APPLET_NOFORK(kill,     kill, BB_DIR_BIN,      BB_SUID_REQUIRE, kill))
 //                   APPLET_NOFORK:name      main  location         suid_type     help
-//applet:IF_KILLALL( APPLET_NOFORK(killall,  kill, BB_DIR_USR_BIN,  BB_SUID_DROP, killall))
-//applet:IF_KILLALL5(APPLET_NOFORK(killall5, kill, BB_DIR_USR_SBIN, BB_SUID_DROP, killall5))
+//applet:IF_KILLALL( APPLET_NOFORK(killall,  kill, BB_DIR_USR_BIN,  BB_SUID_REQUIRE, killall))
+//applet:IF_KILLALL5(APPLET_NOFORK(killall5, kill, BB_DIR_USR_SBIN, BB_SUID_REQUIRE, killall5))
 
 //kbuild:lib-$(CONFIG_KILL) += kill.o
 //kbuild:lib-$(CONFIG_KILLALL) += kill.o
